jerry-core/api/jerryscript.c

```diff
@@ -5645,6 +5645,64 @@ jerry_free_source_info (jerry_source_info_t *source_info_p) /**< source info blo
 #endif /* JERRY_FUNCTION_TO_STRING */
 } /* jerry_free_source_info */
 
+/**
+ * Gets the resource name and location info assigned to a SyntaxError object generated by the parser.
+ *
+ * @return resource name, if a location info is available
+ *         error, otherwise
+ */
+jerry_value_t
+jerry_get_syntax_error_location (jerry_value_t value, /**< SyntaxError object */
```

Comment:
How about not making this syntax error specific, and changing the name to be generic for errors. Functionally it already seems to be generic. Also in the future we could possibly include location info for other error types as well.

Same for the `jerry_syntax_error_location_t` type, I would just call it `jerry_error_location_t`.